.github/files/list-changed-projects.php

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Tool to list whether projects have changed.
+ *
+ * @package automattic/jetpack
+ */
+
+// phpcs:disable WordPress.WP.AlternativeFunctions, WordPress.PHP.DiscouragedPHPFunctions, WordPress.Security.EscapeOutput.OutputNotEscaped
+
+// Files that mean all tests should be run.
+// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+$infrastructure_files = array(
+	'.github/files/list-changed-projects.php',
+	'.github/files/generate-ci-matrix.php',
+	'.github/files/process-coverage.sh',
+	'.github/files/setup-wordpress-env.sh',
+	'.github/php-version',
+	'.github/workflows/tests.yml',
+	'.nvmrc',
+);
+
+// Globs that should be ignored. Supports `*`, `**`, and `?`.
+// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+$ignore_globs = array(
+	'**.md',
+);
+
+chdir( __DIR__ . '/../../' );
+
+if ( array_search( '--debug', $argv, true ) !== false ) {
+	/**
+	 * Output debug info.
+	 *
+	 * @param array ...$args Arguments to printf. A newline is automatically appended.
+	 */
+	function debug( ...$args ) {
+		$args[0] = "\e[1;30m${args[0]}\e[0m\n";
+		fprintf( STDERR, ...$args );
+	}
+} else {
+	/**
+	 * Do not output debug info.
+	 */
+	function debug() {
+	}
+}
+
+/**
+ * Fetch the list of all projects.
+ *
+ * @return string[]
+ */
+function get_all_projects() {
+	static $cache = null;
+
+	if ( null === $cache ) {
+		$cache = array( 'monorepo' );
+		foreach ( glob( 'projects/*/*/composer.json' ) as $file ) {
+			$cache[] = substr( $file, 9, -14 );
+		}
+		ksort( $cache );
+	}
+
+	return $cache;
+}
+
+/**
+ * Fetch the list of changed projects.
+ *
+ * @return string[]
+ */
+function get_changed_projects() {
+	global $infrastructure_files, $ignore_globs;
+
+	// Code coverage probably needs to run all tests to avoid confusing metrics.
+	// @todo Determine if that's actually true.
```

Comment:
Looks like codecov.io can handle [carrying forward from past runs](https://docs.codecov.io/docs/carryforward-flags), but codeclimate.com can't.

OTOH, we might be able to use something like [this action](https://github.com/marketplace/actions/download-base-artifact) to try to download a coverage artifact from a past run and copy it into the coverage directory, if we really want.